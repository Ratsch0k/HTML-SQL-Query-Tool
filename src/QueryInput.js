import React from 'react';
import * as ReactDOM from "react-dom";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import InputGroup from "react-bootstrap/InputGroup";

class QueryInput extends React.Component {
    constructor(props, context) {
        super(props, context);

        // Bind functions to object
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.handleRequest = this.props.onRequest;

        // Initialize state
        this.state = {
            size: 'lg',
            first: true,
            className: 'mb-3 mt-5',
            value: '',
        };
    }

    // Handle submit of sql command
    handleSubmit(event) {
        // Prevent site from refreshing
        event.preventDefault();

        this.handleRequest(this.state.value);

        // If first submit, change state so animation plays
        if (this.state.first) {
            this.setState({size: '', first: false, className: 'mb-1 mt-1'});
        }

        // Reset input
        this.setState({value: ''});
    }
    // Handles user input in query textfield
    handleChange(event){
        // Set state to input
        this.setState({value: event.target.value});
    }

    render() {
        return(
            <Form onSubmit={this.handleSubmit}>
                <InputGroup className={this.state.className} size={this.state.size} id='query-input'>
                    <Form.Control id='query-command'
                                  placeholder='PostgreSQL Query Command'
                                  aria-describedby='query-button'
                                  value={this.state.value}
                                  onChange={this.handleChange}
                                  autoFocus/>
                    <InputGroup.Append>
                        <Button variant='outline-success'
                                onClick={this.handleSubmit} type="button">Run</Button>
                    </InputGroup.Append>
                </InputGroup>
            </Form>
        );
    }
}

export default QueryInput;