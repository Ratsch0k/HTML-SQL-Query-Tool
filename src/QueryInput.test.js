import React from 'react';
import ReactDom from 'react-dom';
import QueryInput from './QueryInput';
import renderer from 'react-test-renderer';

it('QueryInput renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<QueryInput />, div);
    ReactDom.unmountComponentAtNode(div);
});

it('change size when submitted', () => {
    var callbackValue = null;
    const mockUpParentFunction = (value) => {
        callbackValue = value;
    };

    const component = renderer.create(<QueryInput onClick={mockUpParentFunction}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});