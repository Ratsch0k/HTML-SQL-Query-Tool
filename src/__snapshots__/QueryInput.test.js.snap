// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change size when submitted 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="mb-3 mt-5 input-group input-group-lg"
    id="query-input"
  >
    <input
      aria-describedby="query-button"
      autoFocus={true}
      className="form-control"
      id="query-command"
      onChange={[Function]}
      placeholder="PostgreSQL Query Command"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-outline-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Run
      </button>
    </div>
  </div>
</form>
`;
